lib/stm32/common/usart_common_all.c

```diff
@@ -48,9 +48,18 @@ usart_reg_base
 
 void usart_set_baudrate(uint32_t usart, uint32_t baud)
 {
+
+#if defined STM32F4
+	uint32_t clock = rcc_clock_scale_used->apb1_frequency;
+
+	if ((usart == USART1) ||
+	    (usart == USART6)) {
+		clock = rcc_clock_scale_used->apb2_frequency;
+	}
+#else
```

Comment:
Yes a convenience function for getting the uart clock is a good idea. I was more concerned with the RCC part first.